readthedocs/builds/version_slug.py

```diff
@@ -114,13 +104,40 @@ def slugify(self, content):
             space_replacement='-',
         )
 
-        # Remove first character wile it's an invalid character for the
+        # Remove first character while it's an invalid character for the
         # beginning of the slug
-        slugified = slugified.lstrip(self.ok_chars)
+        slug = slug.lstrip(self.ok_chars)
+        slug = slug or self.fallback_slug
+
+        if check_pattern and not self.is_valid(slug):
+            raise Exception(f'Invalid generated slug: {slug}')
```

Comment:
We may want to create our own exception for this: `InvalidVersionSlug` or similar.